aten/src/ATen/gen.py

```diff
@@ -45,6 +45,9 @@
     help='reinterpret CUDA as ROCm/HIP and adjust filepaths accordingly')
 options = parser.parse_args()
 gen_to_source = os.environ.get('GEN_TO_SOURCE')  # update source directly as part of gen
+if BUILD_NAMEDTENSOR:
+    # see Note [Named tensor build codegen]
+    gen_to_source = True
```

Comment:
you should just change the default so it's true if BUILD_NAMEDTENSOR is on -- you should still be able to turn it off explicitly.